server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -696,9 +700,10 @@ public void sendSearchResponse(InternalSearchResponse internalSearchResponse, At
                     searchContextId = null;
                 }
             }
+            onPhaseEnd();
+            onRequestEnd();
             listener.onResponse(buildSearchResponse(internalSearchResponse, failures, scrollId, searchContextId));
         }
-        onPhaseEnd();
```

Comment:
This change is related to a separate pending PR. It will go away once https://github.com/opensearch-project/OpenSearch/pull/10934 is merged and I rebase.